### PR TITLE
Save spline to CSV

### PR DESCRIPTION
**Pull request by zakmat**
This PR enhances the capability of the Spline Tools Component by adding an option to save existing Spline into the CSV file

### CODE REVIEW

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp [hunk 1]
```diff
@@ -63,6 +68,40 @@ namespace SplineTools
...
+        {
+            AZ_Error("SplineToolsEditorComponent", false, "Vertices not saved");
+            return;
+        }
+
+        AZ::Transform worldTM(AZ::Transform::Identity());
+        if (!m_isLocalCoordinates)
+        {
+            AZ::TransformBus::EventResult(worldTM, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+        }
```
Comment (jhanca-robotecai) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp:100]: `worldTM` variable is not used within the code.

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp [hunk 2]
```diff
@@ -63,6 +68,40 @@ namespace SplineTools
...
+        AssetSysReqBus::BroadcastResult(
+            ok, &AssetSysReqBus::Events::GetSourceInfoBySourceUUID, m_csvAssetId.m_guid, sourceAssetInfo, watchFolder);
+        const AZ::IO::Path sourcePath = AZ::IO::Path(watchFolder) / AZ::IO::Path(sourceAssetInfo.m_relativePath);
+
+        AZ_Printf("SplineToolsEditorComponent", "Save csv asset to %s", sourcePath.c_str());
+
+        bool result = SetSplinePointsToCsv(sourcePath.c_str(), vertices);
+        if (!result)
+        {
+            AZ_Error("SplineToolsEditorComponent", false, "Vertices not saved");
```
Comment (jhanca-robotecai) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.cpp:92]: Is this _error_ necessary? There are some _warning_ messages triggered in `SetSplinePointsToCsv`.

File: Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.h [hunk 3]
```diff
@@ -43,5 +43,7 @@ namespace SplineTools
 
         void ReloadCSVAsset();
         AZStd::vector<AZ::Vector3> GetSplinePointsFromCsv(const AZStd::string& csvFilePath);
+        void SaveCsvAsset();
+        bool SetSplinePointsToCsv(const AZStd::string& csvFilePath, const AZStd::vector<AZ::Vector3>& vertices);
```
Comment (jhanca-robotecai) [inline Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.h:47]: Maybe `SaveSplinePointsToCsv` would be a better name?

Comment (zakmat) [reply Gems/RobotecSplineTools/Code/Source/Tools/SplineToolsEditorComponent.h]: Changed to `SavePointsToCsv`

**Comment by zakmat** [comment]
I have also fixed a transformation bug when CSV is filled with world coordinates.

**Review by michalpelka — APPROVED** [review]

**Review by knicked — APPROVED** [review]
Works fine.

**Pull request merged**